pkg/cmd/util/clientcmd/factory.go

```diff
@@ -277,11 +281,18 @@ func (c *clientCache) ClientConfigForVersion(version string) (*kclient.Config, e
 		c.defaultConfig = config
 	}
 	// TODO: have a better config copy method
+	if config, ok := c.configs[version]; ok {
+		return config, nil
+	}
+	// TODO: have a better config copy method
 	config := *c.defaultConfig
-	if len(version) != 0 {
-		config.Version = version
+	negotiatedVersion, err := kclient.NegotiateVersion(nil, &config, version, latest.Versions)
```

Comment:
I had to make one more fix, which was to set the openshift defaults on the client config used to build the negotiator client. I verified that the unversioned path prefix is `/oapi`.